storage/aptosdb/src/ledger_db/event_db.rs

```diff
@@ -190,24 +190,36 @@ impl EventDb {
         start: Version,
         end: Version,
         db_batch: &SchemaBatch,
-    ) -> anyhow::Result<()> {
+        split_indexer_deletes: bool,
+    ) -> anyhow::Result<SchemaBatch> {
         let mut current_version = start;
+        let indexer_batch = SchemaBatch::new();
         for events in self.get_events_by_version_iter(start, (end - start) as usize)? {
             for (idx, event) in (events?).into_iter().enumerate() {
                 if let ContractEvent::V1(v1) = event {
-                    db_batch.delete::<EventByVersionSchema>(&(
-                        *v1.key(),
-                        current_version,
-                        v1.sequence_number(),
-                    ))?;
-                    db_batch.delete::<EventByKeySchema>(&(*v1.key(), v1.sequence_number()))?;
+                    if split_indexer_deletes {
```

Comment:
nit: can be made shorter
```
let index_batch_ref = if let !delete_indices {
	None
} else if internal_indexer_enabled {
       Some(&mut indexer_batch)
} else {
      Some(db_batch)
};
for ..
   for ..
     if let..
         if let(index_batch) = index_batch {
             batch.delete()
             batch.delete()
         }
```

or do the logic outside and pass in `indices_batch: Option<&mut SchemaBatch>`